src/types.ts

```diff
@@ -12,6 +12,31 @@ export interface ClockOptions {
   timezoneSettings: TimezoneSettings;
 }
 
+export enum ClockSource {
+  input = 'input',
+  query = 'query',
+}
+
+export enum CountdownQueryCalculation {
```

Comment:
these two types are almost the same. extract the common part. e.g; `QueryCalculation`